homeassistant/components/upb/const.py

```diff
@@ -0,0 +1,33 @@
+"""Support the UPB PIM."""
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+CONF_NETWORK = "network"
+DOMAIN = "upb"
+
+ATTR_BLINK_RATE = "blink_rate"
+ATTR_BRIGHTNESS = "brightness"
+ATTR_BRIGHTNESS_PCT = "brightness_pct"
+ATTR_RATE = "rate"
+VALID_BRIGHTNESS = vol.All(vol.Coerce(int), vol.Clamp(min=0, max=255))
+VALID_BRIGHTNESS_PCT = vol.All(vol.Coerce(float), vol.Range(min=0, max=100))
+VALID_RATE = vol.All(vol.Coerce(float), vol.Clamp(min=-1, max=3600))
+
+UPB_BRIGHTNESS_RATE_SCHEMA = vol.All(
+    cv.has_at_least_one_key(ATTR_BRIGHTNESS, ATTR_BRIGHTNESS_PCT),
+    cv.make_entity_service_schema(
+        {
+            vol.Exclusive(ATTR_BRIGHTNESS, ATTR_BRIGHTNESS): VALID_BRIGHTNESS,
+            vol.Exclusive(ATTR_BRIGHTNESS_PCT, ATTR_BRIGHTNESS): VALID_BRIGHTNESS_PCT,
+            vol.Optional(ATTR_RATE, default=-1): VALID_RATE,
+        }
+    ),
+)
+
+UPB_BLINK_RATE_SCHEMA = {
+    vol.Required(ATTR_BLINK_RATE, default=0.5): vol.All(
+        vol.Coerce(float), vol.Range(min=0, max=4.25)
+    )
```

Comment:
Shouldn't `entity_id` be part of this service schema?

Reply:
Or even better, just pass the dict inside the schema as the `schema` parameter in `platform.async_register_entity_service`.

Reply:
Should `entity_id` be a part? My understanding is that `async_register_entity_service` does that for you. When looking at other services registered this way I was also curious why `entity_id` was not part of the service, so I looked at the code. Here is where it is added:

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity_platform.py#L507-L508

I'm not saying that what I've got is the right pattern but the `blink` service works. Don't quote me where I learned this, but pretty sure it was the Sonos services here:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/sonos/media_player.py#L253-L263

As for why not pass it as the `schema` param, the next PR for `scene` support reuses the schema.

Happy to change if there is a better way. Most of what I do is by looking at other code and trying to replicate.

Reply:
Sorry, I missed that we're not wrapping this in `vol.Schema`. We usually either define a constant with `vol.Schema` or pass the dict schema inline without the constant.